pyproject.toml

```diff
@@ -15,6 +15,9 @@ markers = [
     "requires_clean_slate: requires a clean slate, so no parallel testing",
 ]
 
+[tool.ruff]
+target-version = "py39"  # Same as python_requires in setup.cfg
```

Comment:
This prevents ruff from trying to replace `from typing_extensions import TypeAlias` with `from typing import TypeAlias`. Strangely, you still need to disable the `UP007` and `UP045` rules, even though they should only apply for python>=3.10

Reply:
When do drop 3.9 support, you'll want to change this too. Ruff could infer it from pyproject.toml settings, but looks like it doesn't try to read it from setup.cfg.